console/console-init/ui/src/modules/iot-device/components/AddGateways/AddGateway.tsx

```diff
@@ -15,34 +15,80 @@ import {
   Chip,
   ChipGroup,
   GridItem,
-  Grid
+  Grid,
+  Popover,
+  DropdownPosition,
+  ValidatedOptions
 } from "@patternfly/react-core";
-import { DeviceListAlert } from "modules/iot-device/components";
 import { deviceIDRegExp } from "types/Configs";
+import { InfoCircleIcon } from "@patternfly/react-icons";
+import { DropdownWithToggle, TypeAheadSelect } from "components";
+import "./pf-overrides.css";
+import { RETURN_ALL_DEVICES_FOR_IOT_PROJECT } from "graphql-module/queries";
+import { getInitialFilter, gatewayTypeOptions } from "modules/iot-device/utils";
+import { IIoTDevicesResponse } from "schema";
+import { useApolloClient, useQuery } from "@apollo/react-hooks";
 
 export interface IGatewaysProps {
   header?: string;
   gateways?: string[];
+  gatewayGroups?: string[];
   returnGateways?: (gateways: string[]) => void;
+  returnGatewayGroups?: (gatewayGroups: string[]) => void;
 }
 
 export const AddGateways: React.FunctionComponent<IGatewaysProps> = ({
   header = "Add permitted devices as gateways to this new device",
   gateways: gatewayList = [],
-  returnGateways
+  gatewayGroups: groupList = [],
+  returnGateways,
+  returnGatewayGroups
 }) => {
+  const client = useApolloClient();
   const [inputID, setInputID] = useState<string>();
   const [isValid, setIsValid] = useState<boolean>(false);
   const [gateways, setGateways] = useState<string[]>(gatewayList);
+  const [gatewayGroups, setGatewayGroups] = useState<string[]>(groupList);
+  const [inputType, setInputType] = useState<string>("Device ID");
+  const [selectedGateways, setSelectedGateways] = useState<any[]>([]);
 
   useEffect(() => {
     returnGateways && returnGateways(gateways);
   }, [gateways]);
 
-  const addGateway = () => {
+  useEffect(() => {
+    returnGatewayGroups && returnGatewayGroups(gatewayGroups);
+  }, [gatewayGroups]);
+
+  const addGateway = async () => {
     if (inputID) {
-      setGateways([...gateways, inputID]);
-      setInputID("");
+      const filterObj = getInitialFilter();
+      filterObj.deviceId = inputID;
+
+      const { data } = await client.query<IIoTDevicesResponse>({
+        query: RETURN_ALL_DEVICES_FOR_IOT_PROJECT(
+          1,
```

Comment:
why do we have hardcode values here ?

Reply:
Cause we are just checking if the device with ID exists or not, for that we are passing page = 1, and perpage = 1.